### DISCUSSION TITLE
Raw link token storage

### DISCUSSION THREAD

**Discussion by Vimal0008** [community/community#35885] [Code Security]
Hi Team,

I had created a private repositories and tried to call the url through SAS studio. But each time when I'm opening the link it is giving new tokens attached to it. So can you help me to generate static token for raw links or help me with another approach where I can handle this issue?

Your help will be highly appreciated.

**Comment by airtower-luna** [comment]
The URL tokens expire after a while, probably to make sure someone who got a raw link to a private repository won't have access forever without the owner noticing.

The alternative you have is to use a link without a token, and [send an `Authorization` header](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#authentication). You'll need a valid OAuth token to construct the header, a PAT works.
